### PR TITLE
Add unit test for Vector2 `plane_project`

### PR DESCRIPTION
**Pull request by Nazarwadim**

### CODE REVIEW

File: tests/core/math/test_vector2.h [hunk 1]
```diff
@@ -372,6 +373,9 @@ TEST_CASE("[Vector2] Plane methods") {
 	CHECK_MESSAGE(
 			vector.project(vector_normal).is_equal_approx(Vector2(2.0292559899117276166, 0.60126103404791929382)),
 			"Vector2 projected on a normal should return expected value.");
+	CHECK_MESSAGE(
+			vector.plane_project(p_d, vector_normal).is_equal_approx(Vector2(117.339048445267494, 330.028130938608797)),
+			"Vector2 plane proeject should return expected value.");
```
Comment (Geometror) [inline tests/core/math/test_vector2.h:378]:
```suggestion
	CHECK_MESSAGE(
			vector_normal.plane_project(p_d, vector).is_equal_approx(Vector2(8.75944, 5.638667)),
			"Vector2 plane proeject should return expected value.");
```
The plane should be defined via a normal and a distance, the parameter of this method is actually the vector to project. The values above are with p_d set to 10 (validated visually).

File: tests/core/math/test_vector2.h [hunk 2]
```diff
@@ -372,6 +373,9 @@ TEST_CASE("[Vector2] Plane methods") {
 	CHECK_MESSAGE(
 			vector.project(vector_normal).is_equal_approx(Vector2(2.0292559899117276166, 0.60126103404791929382)),
 			"Vector2 projected on a normal should return expected value.");
+	CHECK_MESSAGE(
+			vector_normal.plane_project(p_d, vector).is_equal_approx(Vector2(94.187635516479631, 30.951892004882851)),
+			"Vector2 plane proeject should return expected value.");
```
Comment (Maran23) [inline tests/core/math/test_vector2.h:378]: Typo: `proeject` should be `project`.

File: tests/core/math/test_vector2.h [hunk 3]
```diff
@@ -372,6 +373,9 @@ TEST_CASE("[Vector2] Plane methods") {
 	CHECK_MESSAGE(
 			vector.project(vector_normal).is_equal_approx(Vector2(2.0292559899117276166, 0.60126103404791929382)),
 			"Vector2 projected on a normal should return expected value.");
+	CHECK_MESSAGE(
+			vector_normal.plane_project(p_d, vector).is_equal_approx(Vector2(94.187635516479631, 30.951892004882851)),
+			"Vector2 plane project should return expected value.");
```
Comment (AThousandShips) [inline tests/core/math/test_vector2.h:378]:
```suggestion
			"Vector2 plane_project should return expected value.");
```

**Review by AThousandShips — APPROVED** [review]

**Review by Maran23 — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**